### DISCUSSION TITLE
Tracking the bodies in raycasts

### DISCUSSION THREAD

**Discussion by LeXXik** [jrouwe/JoltPhysics.js#55] [Q&A] [answered]
We have `collector.OnBody` and `collector.AddHit`. Their order of execution is not interleaved, so in case of multiple bodies, we get something like:
```
.OnBody()
.OnBody()
.OnBody()

.AddHit()
.AddHit()
.AddHit()
```

Curious how do you guys keep track which body was hit in `AddHit()`. If I add them to the array during `.OnBody()`, will the order be the same, when I access them in `.AddHit()` (e.g. would the second .AddHit be referring to the second .OnBody)?

**Comment by LeXXik** [comment]
Actually, it is interleaved. Hmm. I am getting more .AddHit, than .OnBody. Probably need to dig sources to understand what is calling .AddHit without .OnBody.

**Comment by LeXXik** [comment, marked as answer]
I see, each body the ray hits calls .OnBody, with subsequent .OnHit for each surface of that body.

**Reply by jrouwe** [reply]
Yes, that's correct.
